base/src/types.rs

```diff
@@ -351,6 +351,16 @@ pub enum FontScheme {
     None,
 }
 
+// Merge Cell type
+// There will be one MergeCell maintained for every Merge cells that we load
+// It will consist of range_ref : its range reference in String representation
+// merge_cell_range : Its tuple having [row_start, column_start, row_end, column_end]
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+pub struct MergeCell {
+    pub merge_cell_range: (i32, i32, i32, i32),
+    pub range_ref: String,
+}
+
```

Comment:
I would suggest some slightly different naming:

```rust
pub struct MergedRange {
  range: (i32, i32, i32, i32),
  r#ref: String
}
```

It is not a cell in it a range of cells that is merged. Also the thing is a _merged_ range not a _merge_ range but we _merge_ a range of cells into a _merged_ range.

Second thing. Do we need the 'ref' at all? It is redundant, opening the door to bugs where the range and the ref fall out of sync. We could use it if we would need it for perfomance reasons. another reason to keep the string representation is if it differs of what can be computed. For instance if the user enters `a1:a3` but we would compute `A1:A3`. This is not the case in here.

I would just try:
```rust
type MergedRange = (i32, i32, i32, i32)
```

and provide some utility function to compute the str representation.

Reply:
Note that `ref` is a keyword in Rust. You can use keywords in Rust by escaping them with `r#` in front of them.